### PR TITLE
ci: add and adjust labels to autotranslate PRs

### PR DESCRIPTION
**Pull request by rexxars**
This PR does three things:

1. Reverts the changes done to the autotranslate CI workflow attempting to get more granular permissions. I do not have time to investigate it fully to understand why it is failing, so for now we'll just roll with what we had (which worked).
2. Adds `autotranslate` label for new auto translate PRs, as well as either a `awaiting-review` label (when there are maintainers we can ping) or `maintainerless` when there are no maintainers for a language. These PRs should probably just be automerged, since we usually don't really have any input on them, but will leave that for another day.
3. Adjusts the labels on PR reviews (accepted and changes requested).  If a PR is approved by a maintainer, an `approved` label will be added and the `awaiting-review` will be removed. If there are requested changes, the `awaiting-review` is removed, along with any potential `approved` label (in the case of maintainers disagreeing).

This should help us see the status of PRs at a glance. In the future we may consider automatically merging approved PRs after a given time, or if there is only a single maintainer which has approved it.

### CODE REVIEW

**Review by runeb — APPROVED** [review]
LGTM

edit: fails here btw https://github.com/sanity-io/locales/actions/runs/7562298114/job/20592477142?pr=279#step:6:14

**Comment by runeb** [comment]
@rexxars should we wait to merge this till after auto PR branches have merged because of the github action task on package changes?

**Comment by rexxars** [comment]
> edit: fails here btw https://github.com/sanity-io/locales/actions/runs/7562298114/job/20592477142?pr=279#step:6:14

Oops. Thanks. Shouldn't run for all branches, only auto-PRs. Fixed that now. Also corrected access to the PR number (I think).

> @rexxars should we wait to merge this till after auto PR branches have merged because of the github action task on package changes?

Merging this may force push to the auto branches, but all the currently open ones are unedited and awaiting review - so it _shouldn't_ be a problem?

**Pull request merged**
